src/wallet/rpcwallet.cpp

```diff
@@ -1059,27 +1077,38 @@ static UniValue ListReceived(const CWallet* const pwallet, const UniValue& param
 
     bool has_filtered_address = false;
     CTxDestination filtered_address = CNoDestination();
-    if (!by_label && params.size() > 3) {
+    if (!by_label && !params[3].isNull() && !params[3].get_str().empty()) {
         if (!IsValidDestinationString(params[3].get_str())) {
             throw JSONRPCError(RPC_WALLET_ERROR, "address_filter parameter was invalid");
         }
         filtered_address = DecodeDestination(params[3].get_str());
         has_filtered_address = true;
     }
 
+    // getreceivedbyaddress has address filter at 3rd index, so next params index is 4
+    // getreceivedbylabel has no filter parameter, so next params index is 3
+    int next_index{by_label ? 3 : 4};
+    bool include_immature{params[next_index].isNull() ? false : params[next_index].get_bool()};
+
+    // Filtering out coinbase outputs is deprecated
+    // It can be enabled by setting deprecatedrpc=filter_coinbase
+    bool filter_coinbase{pwallet->chain().rpcEnableDeprecated("filter_coinbase")};
+
     // Tally
     std::map<CTxDestination, tallyitem> mapTally;
     for (const std::pair<const uint256, CWalletTx>& pairWtx : pwallet->mapWallet) {
         const CWalletTx& wtx = pairWtx.second;
 
-        if (wtx.IsCoinBase() || !pwallet->chain().checkFinalTx(*wtx.tx)) {
-            continue;
-        }
-
         int nDepth = wtx.GetDepthInMainChain();
         if (nDepth < nMinDepth)
             continue;
 
+        // Coinbase with less than 1 confirmation is an orphan
+        if ((wtx.IsCoinBase() && (nDepth < 1 || filter_coinbase))
+            || (!include_immature && wtx.IsImmatureCoinBase())
+            || !pwallet->chain().checkFinalTx(*wtx.tx))
+            continue;
```

Comment:
Consider enclosing the `continue;` within braces? I'm aware they weren't there previously, but when the condition becomes multi-line, perhaps they help with readability.

Reply:
Thanks for the review! I've updated with your suggestions along with the latest rebase.